addon/record-data.js

```diff
@@ -497,7 +511,33 @@ export default class M3RecordData {
   }
 
   isDeletionCommitted() {
-    return this._isDeletionCommited;
+    if (this._baseRecordData) {
+      return this._baseRecordData.isDeletionCommitted();
+    }
+    return this._isDeletionCommitted;
+  }
+
+  setIsDeletionCommitted(value) {
+    if (this._baseRecordData) {
+      return this._baseRecordData.setIsDeletionCommitted(value);
+    }
+    this._isDeletionCommitted = value;
+    this._notifyStateChange();
+  }
+
+  _notifyStateChange() {
+    let record = recordDataToRecordMap.get(this);
+    if (this.isDeletionCommitted()) {
+      this.removeFromRecordArrays();
+    }
+    if (record) {
+      record._updateCurrentState();
+    }
+    if (this._projections) {
+      for (let i = 1; i < this._projections.length; i++) {
+        this._projections[i]._notifyStateChange();
+      }
```

Comment:
why do we skip `projections[0]`? that's the base (unprojected) case?

Reply:
> that's the base (unprojected) case? -> Yup, it's a pattern we use in a few places I think

We always notify on the base case itself, so 
```
    let record = recordDataToRecordMap.get(this);
    if (this.isDeletionCommitted()) {
      this.removeFromRecordArrays();
    }
    if (record) {
      record._updateCurrentState();
    }
```
deals with the base and 
```
      for (let i = 1; i < this._projections.length; i++) {
        this._projections[i]._notifyStateChange();
      }
      ```
      with the projections

Reply:
yeah it's unfortunate that we have to remember it at every loop of `this._projections` but oh well :shrug: